_posts/2021-01-27-#20833.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: pr
+date: 2021-01-27
+title: "Enable packages through testmempoolaccept"
+pr: 20833 
+authors: [glozow]
+components: ["rpc/rest/zmq", "tests", "validation"]
+host: glozow
+status: upcoming
+commit: 6574758
+---
+## Definitions
+
+- A **package** is a group of transactions that have some dependency relationship, i.e., some
+transaction spends the outputs of another transaction in the package.
+We might call a package a "transaction chain" when it has multiple generations of such dependencies.
+
+- A **test accept** of validation refers to an invocation of `MemPoolAccept` that validates but
+does not submit a transaction to the mempool, called from the `testmempoolaccept` RPC.
+
+- **Coins** ([`CCoins` class](https://doxygen.bitcoincore.org/struct_c_coin.html)) are UTXOs.
+Note that, in our context, coins don't necessarily need to exist on-chain; they can be outputs from
+transactions in the mempool or some other valid but unconfirmed transaction.
+A **CoinsView** is a data structure that contains information about the available coins.
+
+## Background
+
+- The `testmempoolaccept` RPC was introduced in
+[PR#11742](https://github.com/bitcoin/bitcoin/pull/11742) and allows users to test raw transactions
+for validity.
+On master, `testmempoolaccept` is limited to one transaction at a time, which means users cannot
+test transactions that depend on outputs that the mempool has not seen yet.
+The obstacle is that our validation code uses a `CoinsViewMemPool`, which tracks the coins
+available in the current chain state and mempool (and not anything else), to validate transaction
+inputs.
+
+- Another reason package mempool accept may be of interest is improving fee bumping.
+Currently, Child Pays for Parent does not work for transactions with a feerate below the mempool
+minimum fee, which can be dangerous for recipients of time-sensitive transactions.
+[PR#20833](https://github.com/bitcoin/bitcoin/pull/20833) is a first step towards package
+validation and hopefully, eventually, Package Relay.
```

Comment:
Suggestion: link to https://bitcoincore.reviews/16401, https://github.com/bitcoin/bitcoin/issues/14895 and https://gist.github.com/sdaftuar/8756699bfcad4d3806ba9f3396d4e66a

Reply:
I'll link to the #16401 review club and optech topic package relay?